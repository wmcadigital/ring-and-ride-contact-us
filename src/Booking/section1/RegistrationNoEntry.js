import PropTypes from "prop-types";
import { useFormState } from "react-final-form";
import { useNavigate } from "react-router-dom";

import FormSection from "../../common/FormSection";
import TextInput from "../../common/TextInput";
import ProgressIndicator from "../../common/ProgressIndicator";
import Question from "../../common/Question";
import ButtonLink from "../../common/ButtonLink";
import {
  composeValidators,
  required,
  registrationNumber,
  registrationNumberLength,
} from "../../common/validation";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutSectionName from "./getAboutSectionName";

const RegistrationNoEntry = ({ orderNo }) => {
  const stateApi = useFormState();

  const navigate = useNavigate();

  const formValues = stateApi.values;
  const question =
    formValues["bookingParty"] === "behalfSomeone"
      ? `What is ${formValues["firstName"]}'s registration number?`
      : "What is your registration number";

  const error = stateApi.submitFailed ? stateApi.errors?.registrationNo : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("1", stateApi)}
        sectionName={getAboutSectionName(stateApi)}
      />
      <Question text={question} />
      <TextInput
        fieldName="registrationNo"
        label={
          <>
            {" "}
            Registration number
            <p>For example, 123456</p>
          </>
        }
        validation={composeValidators(
          required,
          registrationNumber,
          registrationNumberLength
        )}
        error={error}
      />
      <ButtonLink
        callback={() =>
          navigate("/booking/forgotRegistration", {
            state: { formValues, orderNo },
            replace: true,
          })
        }
      >
        I don&apos;t remember my registration number
      </ButtonLink>
    </FormSection>
  );
};

RegistrationNoEntry.propTypes = {
  orderNo: PropTypes.number,
};

export default RegistrationNoEntry;
