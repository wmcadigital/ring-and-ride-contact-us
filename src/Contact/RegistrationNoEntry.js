import { useFormState } from "react-final-form";

import TextInput from "../common/TextInput";
import ProgressIndicator from "../common/ProgressIndicator";
import Question from "../common/Question";
import { numbersOnly } from "./validation";

const RegistrationNoEntry = () => {
  const stateApi = useFormState();

  const error = stateApi.submitFailed ? stateApi.errors?.registrationNo : null;

  return (
    <div className="wmrards-fe-group wmrards-m-t-20">
      <ProgressIndicator
        sectionPosition="Section 2 of 2"
        sectionName="About you"
      />
      <Question text="Do you have a Ring and Ride registration number?" />
      <TextInput
        fieldName="registrationNo"
        label={
          <>
            {" "}
            Registration number
            <p>For example, 1234567890</p>
          </>
        }
        validation={numbersOnly}
        error={error}
      />
    </div>
  );
};

export default RegistrationNoEntry;
