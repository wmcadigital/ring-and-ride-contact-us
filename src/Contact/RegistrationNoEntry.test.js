import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import RegistrationNoEntry from "./RegistrationNoEntry";

jest.mock("react-final-form");

describe("RegistrationNoEntry", () => {
  it("renders with no value", () => {
    useFormState.mockImplementationOnce(() => ({}));
    const renderer = createRenderer();

    renderer.render(<RegistrationNoEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with a value", () => {
    useFormState.mockImplementationOnce(() => ({
      values: { registrationNo: "12345" },
    }));
    const renderer = createRenderer();

    renderer.render(<RegistrationNoEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      submitFailed: true,
      errors: { registrationNo: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<RegistrationNoEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
