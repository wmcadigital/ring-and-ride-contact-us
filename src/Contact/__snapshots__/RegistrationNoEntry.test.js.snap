// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationNoEntry renders with a value 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="Do you have a Ring and Ride registration number?"
  />
  <TextInput
    error={null}
    fieldName="registrationNo"
    label={
      <React.Fragment>
         
        Registration number
        <p>
          For example, 1234567890
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;

exports[`RegistrationNoEntry renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="Do you have a Ring and Ride registration number?"
  />
  <TextInput
    error="Required"
    fieldName="registrationNo"
    label={
      <React.Fragment>
         
        Registration number
        <p>
          For example, 1234567890
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;

exports[`RegistrationNoEntry renders with no value 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="Do you have a Ring and Ride registration number?"
  />
  <TextInput
    error={null}
    fieldName="registrationNo"
    label={
      <React.Fragment>
         
        Registration number
        <p>
          For example, 1234567890
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;
